ui/panels-plugin/src/plugins/time-series-chart/utils/palette-gen.ts

```diff
@@ -11,39 +11,57 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { PaletteOptions } from '../time-series-chart-model';
+import ColorHash from 'color-hash';
 
 /**
- * Helper function to generate a random color for a chart series based on its name
+ * Get color from generative color palette
  */
-export function getRandomColor(identifier: string): string {
-  let hash = 0;
-  for (let index = 0; index < identifier.length; index++) {
-    hash = identifier.charCodeAt(index) + ((hash << 5) - hash);
-  }
-  // Use HSLA to only get random "bright" colors from this
-  const color = `hsla(${~~(180 * hash)},50%,50%,0.8)`;
-  return color;
+export function getAutoPaletteColor(name: string, fallbackColor: string): string {
+  // corresponds to 'Auto' in palette.kind for generative color palette
+  const generatedColor = getConsistentSeriesNameColor(name);
+  return generatedColor ?? fallbackColor;
 }
 
 /**
  * Get line color as well as color for tooltip and legend, account for whether palette is 'Cateogrical' or 'Auto' (generative)
  */
-export function getSeriesColor(
-  name: string,
-  seriesIndex: number,
-  palette: string[],
-  paletteKind: PaletteOptions['kind'] = 'Auto'
-): string {
-  if (paletteKind === 'Categorical' && Array.isArray(palette)) {
-    const colorIndex = seriesIndex % palette.length;
-    // TODO: take fallback color from theme
-    const seriesColor = palette[colorIndex];
-    if (seriesColor !== undefined) {
-      return seriesColor;
-    }
-  }
-
-  // corresponds to 'Auto' in palette.kind
-  return getRandomColor(name);
+export function getCategoricalPaletteColor(palette: string[], paletteIndex: number, fallbackColor: string): string {
+  // Loop through predefined static color palette
+  const colorIndex = paletteIndex % palette.length;
+  // fallback color comes from echarts theme
+  const seriesColor = palette[colorIndex] ?? fallbackColor;
+  return seriesColor;
 }
+
+// Valid hue values are 0 to 360 and can be adjusted to control the generated colors.
+// More info: https://github.com/zenozeng/color-hash#custom-hue
+// Picked min of 30 and max of 360 to exclude common threshold colors (reddish).
+// Series names with "error" in them will always be generated as red.
+const ERROR_HUE_CUTOFF = 30;
+const colorGenerator = new ColorHash({ hue: { min: ERROR_HUE_CUTOFF, max: 360 } });
```

Comment:
Does setting min to `ERROR_HUE_CUTOFF` mean we are excluding most/all red hue from color generator? 

I'm thinking if we should just set `ERROR_HUE_CUTOFF` to a smaller number or dedicate like 5-10 different shades of red for the `error` case,  so we get more of a variety of colors. In your screenshot where all series names include errors, I see some colors that i don't think it's very red.

Reply:
Yeah that's right, I wrote a little bit about this in my other comment [here](https://github.com/perses/perses/pull/1088#discussion_r1171696993) about colorblind friendliness. This is helpful though, I didn't realize some of the error colors weren't red, I reduced the ERROR_HUE_CUTOFF from 30 to 10!